### ISSUE TITLE
CA1508 should not occur in a using statement's expression

### ISSUE THREAD

**Issue by Tan90909090** [dotnet/roslyn-analyzers#4382]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

**Version**: v3.3.1 (Latest)

### Describe the bug
I don't expect CA1508 on using statement's expression.

### Steps To Reproduce
```cs
using System;

class C
{
    void F(IDisposable x)
    {
        if (x != null)
        {
            using (x) { } // CA1508 occurs here
        }
    }

    static void Main() { }
}
```

### Expected behavior
I know that  [a using statement expands to a code containing null checking](https://stackoverflow.com/questions/278902/using-statement-vs-try-finally/278924#278924), but I use a using statement for disposing object. So I don't expect CA1508 on using statement's expression.

### Actual behavior
CA1508 occurs: `'x' is never 'null'. Remove or refactor the condition(s) to avoid dead code.`

## Additional context
Note 1. A variable used for a conditional branch and a variable used for a using statement are irrelevant:
```cs
using System;
using System.IO;

class C
{
    void F(IDisposable x)
    {
        if (x != null)
        {
            using (new MemoryStream()) { } // CA1508 occurs here
        }
    }

    static void Main() { }
}
```

Note 2. If I omit the null checking then CA1508 does not occur in a using statement:
```cs
using System;
using System.IO;

class C
{
    void F(IDisposable x)
    {
        using (new MemoryStream()) { } // CA1508 does NOT occur here
    }

    static void Main() { }
}
```

Note 3. If I use a using statement for `local_variable_declaration` then CA1508 does not occur even if a null checking exists:
```cs
using System;
using System.IO;

class C
{
    void F(IDisposable x)
    {
        if (x != null)
        {
            using (var y = new MemoryStream()) { } // CA1508 does NOT occur here
        }
    }

    static void Main() { }
}
```
